test/e2e/allocator_test.go

```diff
@@ -106,7 +106,7 @@ func TestAllocatorWithDeprecatedRequired(t *testing.T) {
 		helper.ValidateAllocatorResponse(t, response)
 
 		// let's do a re-allocation
-		if runtime.FeatureEnabled(runtime.FeatureStateAllocationFilter) && runtime.FeatureEnabled(runtime.FeaturePlayerAllocationFilter) {
+		if runtime.FeatureEnabled(runtime.FeaturePlayerAllocationFilter) {
 			logrus.Info("testing state allocation filter")
```

Comment:
We should probably pull the state allocation filter tests out of this `if runtime.FeatureEnabled(runtime.FeaturePlayerAllocationFilter)` block.